homeassistant/components/esphome/event.py

```diff
@@ -0,0 +1,48 @@
+"""Support for ESPHome event components."""
+
+from __future__ import annotations
+
+from aioesphomeapi import EntityInfo, Event, EventInfo
+
+from homeassistant.components.event import EventDeviceClass, EventEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.enum import try_parse_enum
+
+from .entity import EsphomeEntity, platform_async_setup_entry
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up ESPHome event based on a config entry."""
+    await platform_async_setup_entry(
+        hass,
+        entry,
+        async_add_entities,
+        info_type=EventInfo,
+        entity_type=EsphomeEvent,
+        state_type=Event,
+    )
+
+
+class EsphomeEvent(EsphomeEntity[EventInfo, Event], EventEntity):
+    """An event implementation for ESPHome."""
+
+    @callback
+    def _on_static_info_update(self, static_info: EntityInfo) -> None:
+        """Set attrs from static info."""
+        super()._on_static_info_update(static_info)
+        static_info = self._static_info
+        if event_types := static_info.event_types:
+            self._attr_event_types = event_types
+        self._attr_device_class = try_parse_enum(
+            EventDeviceClass, static_info.device_class
+        )
+
+    @callback
+    def _on_state_update(self) -> None:
+        self._update_state_from_entry_data()
+        self._trigger_event(self._state.event_type)
```

Comment:
Probably need to double check that this doesn't fire when you reload the integration

Reply:
Yeah looks like it does
![2024-04-24_11-22-54](https://github.com/home-assistant/core/assets/3060199/82361c2f-ec3a-46ca-8542-27d6b040d615)

Reply:
Actually I am not so sure. I have not changed anything but it wont trigger now. Tried in debug mode to to trace the execution to `_on_state_update` but its not being called on reload.

Reply:
AFAICT `_on_state_update` is only ever called when a state message comes from the device which does not happen for an event entity until it is actually triggered so there should be no duplicates.

Reply:
The code looks fine otherwise. If you are confident its not a problem, it seems good to merge